calico.yaml

```diff
@@ -0,0 +1,442 @@
+
+          # TODO: Alternatively, we could patch the unused variable out...
+          export CFLAGS="-O2 -Wall -fomit-frame-pointer -march=x86-64-v2 -Wp,-D_FORTIFY_SOURCE=3"
+          CFLAGS="$CFLAGS -Wno-error=unused-but-set-variable" make -C bpf-apache CC=clang LD=llc
+          CFLAGS="$CFLAGS -Wno-error=unused-but-set-variable" make -C bpf-gpl CC=clang LD=llc
+      # Equivalent to target: "libbpf"
+      - runs: |
+          _arch=$(uname -m); case $_arch in aarch64) _arch="arm64" ;; x86_64) _arch="amd64" ;; esac
+
+          mkdir -p bpf-gpl/include/libbpf/src/$_arch
+          export CFLAGS="-O2 -Wall -fomit-frame-pointer -march=x86-64-v2 -Wp,-D_FORTIFY_SOURCE=3"
+          CFLAGS="$CFLAGS" make -C bpf-gpl/include/libbpf/src BUILD_STATIC_ONLY=2 OBJDIR=$_arch
+  # This fetches and builds the statically compiled calico fork of BIRD
+  - working-directory: bird
+    pipeline:
+      - uses: fetch
+        with:
+          uri: https://github.com/projectcalico/bird/tarball/v0.3.3-202-g7a77fb73
+          expected-sha512: 87d03464163459154dce11952025f4a0de316dbfe0d1f6a3f19a82e951dfca2eb5d0cd18d95eda3abd38d44a9967fbb4b017158d8a5f48951683e3dc87dcd914
+      - runs: |
+          ARCH=${{build.arch}} DIST=dist/ ./create_binaries.sh
+  - working-directory: node
+    pipeline:
+      # Equivalent to target: "filesystem/etc/calico/confd/conf.d"
+      - runs: |
+          _target="filesystem/etc/calico/confd/conf.d"
+          rm -rf $_target && cp -r ../confd/etc/calico/confd/conf.d $_target
+          chmod +w $_target
+      # Equivalent to target: "filesystem/etc/calico/confd/templates"
+      - runs: |
+          _target="filesystem/etc/calico/confd/templates"
+          rm -rf $_target && cp -r ../confd/etc/calico/confd/templates $_target
+          chmod +w $_target
+      # Equivalent to target: "filesystem/usr/lib/calico/bpf"
+      - runs: |
+          _target="filesystem/usr/lib/calico/bpf"
+          rm -rf $_target && mkdir -p $_target
+
+          # This is handled in the felix build pipeline above
+          # make -C ../felix build-bpf ARCH=$(ARCH)
+          cp -r ../felix/bpf-gpl/bin/* $_target
+          cp -r ../felix/bpf-apache/bin/* $_target
+
+subpackages:
+  - name: "calico-node"
+    dependencies:
+      runtime:
+        - runit
+        - iptables
+        - ip6tables
+        - ipset
+        - libpcap
+        - libelf
+        - zlib
+        - libbpf
+        # listed in Dockerfile, but not sure if they're build dependencies (for iptables) or runtime
+        - libmnl
+        - libnfnetlink
+        - libnetfilter_conntrack
+        - iproute2
+        - procps
+        - conntrack-tools
+        - libnetfilter_cthelper
+        - libnetfilter_cttimeout
+        - libnetfilter_queue
+    pipeline:
+      # Equivalent to target: "$(NODE_CONTAINER_BINARY)"
+      - if: ${{build.arch}} == 'x86_64' || ${{build.arch}} == 'aarch64' # Essentially a guard to make sure we never try to build beyond x86_64 and aarch64
```

Comment:
In the event that we add a third supported architecture, what will happen? Will this fail, or will it silently produce an invalid/empty package?

Reply:
I'd be fine if it failed for that arch, since it would give us a hint that we need to do something in order to proceed. Silently producing an empty package is bad, but silently not producing a package could also lead to downstream confusion, if an image expects to use this package and be built for some third arch.

Reply:
In that case, an assert + subpipeline is needed

Reply:
I added an `assertions` guard on the arch specific subpackages. I think that, along with the `target-architecture` in the package config should be good enough 🤞